framework/db/mssql/Schema.php

```diff
@@ -22,6 +22,16 @@ class Schema extends \yii\db\Schema
      */
     public $defaultSchema = 'dbo';
     /**
+     * @var array map of DB errors and corresponding exceptions
+     * If left part is found in DB error message exception class from the right part is used.
+     */
+    public $exceptionMap = [
+        'SQLSTATE[23' => 'yii\db\IntegrityException',
+        'SQLSTATE[HY000]: General error: 20018 The INSERT statement conflicted with the FOREIGN KEY constraint' => 'yii\db\IntegrityException',
```

Comment:
I ran the tests using dblib on Linux. I think we should also run the tests on the sqlsrv Windows driver. I'll try that later.

I guess it can make a difference, when I tried using an odbc driver (with same FreeTDS library as the dblib one) it completely failed on stuff like binding params.